### ISSUE TITLE
Makes two output files.. second one with same name put with "" on the end..

### ISSUE THREAD

**Issue by pjatinsight** [danielebailo/couchdb-dump#43]
Hi,
Thanks for a very useful script.
Just thought you might like to know of some slightly odd results..

I'm on a mac using Yosemite.
The script works great.. but oddly outputs TWO files.

For example: a commandline like this:

bash couchdb-backup.sh -b -H 127.0.0.1 -d my_users -f my_users.json

will produce two files:

 my_users.json
 my_users.json""

The json file produced appears to be just fine..
Its just the two files.. strikes me as odd...

Thanks again..

**Comment by dalgibbard** [comment]
Could you run it with ```bash -x``` and throw us the output please? :)

**Comment by pjatinsight** [comment]
Sure thing, see below:

$ bash -x couchdb-backup.sh -b -H 127.0.0.1 -d bizcomp_users -f bizcomp_users.json
+ scriptversionnumber=1.1.2
+ '[' x-b = x ']'
+ username=
+ password=
+ backup=false
+ restore=false
+ port=5984
+ OPTIND=1
+ lines=5000
+ attempts=3
+ getopts ':h?H:d:f:u:p:P:l:t:a:V?b?B?r?R?' opt
+ case "$opt" in
+ backup=true
+ getopts ':h?H:d:f:u:p:P:l:t:a:V?b?B?r?R?' opt
+ case "$opt" in
+ url=127.0.0.1
+ getopts ':h?H:d:f:u:p:P:l:t:a:V?b?B?r?R?' opt
+ case "$opt" in
+ db_name=bizcomp_users
+ getopts ':h?H:d:f:u:p:P:l:t:a:V?b?B?r?R?' opt
+ case "$opt" in
+ file_name=bizcomp_users.json
+ getopts ':h?H:d:f:u:p:P:l:t:a:V?b?B?r?R?' opt
+ shift 7
+ '[' '' = -- ']'
+ '[' '!' x = x ']'
+ '[' true = true ']'
+ '[' false = true ']'
+ '[' true = false ']'
+ '[' x127.0.0.1 = x ']'
+ '[' xbizcomp_users = x ']'
+ '[' xbizcomp_users.json = x ']'
+ file_name_orig=bizcomp_users.json
++ uname -s
+ os_type=Darwin
+ '[' Darwin = Darwin ']'
++ sysctl -n hw.ncpu
+ cores=8
+ '[' '!' x = x ']'
++ expr 8 - 1
+ threads=7
+ case $attempts in
+ true
++ echo 127.0.0.1
++ grep -c http
+ '[' '!' 0 = 1 ']'
+ url=http://127.0.0.1
++ echo http://127.0.0.1
++ egrep -c ':[0-9]*$'
+ '[' '!' 0 = 1 ']'
+ url=http://127.0.0.1:5984
+ '[' '!' x = x ']'
+ '[' '!' x = x ']'
+ '[' '!' x = x ']'
+ '[' Darwin = Darwin ']'
+ sed_regexp_option=E
+ sed_edit_in_place='-i ""'
++ echo http://127.0.0.1:5984
++ grep -ic '^https://'
+ '[' 0 = 1 ']'
++ which curl
+ '[' x/usr/bin/curl = x ']'
+ '[' true = true ']'
+ '[' false = false ']'
+ '[' -f bizcomp_users.json ']'
+ curl -X GET 'http://127.0.0.1:5984/bizcomp_users/_all_docs?include_docs=true&attachments=true' -o bizcomp_users.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5884    0  5884    0     0  1564k      0 --:--:-- --:--:-- --:--:-- 1915k
+ '[' '!' 0 = 0 ']'
++ head -n 1 bizcomp_users.json
++ grep '^{"error'
+ ERR_CHECK=
+ '[' '!' x = x ']'
++ file bizcomp_users.json
++ grep -c CRLF
+ '[' 1 = 1 ']'
+ echo '... INFO: File contains Windows carridge returns- converting...'
... INFO: File contains Windows carridge returns- converting...
++ du -P -k bizcomp_users.json
++ awk '{print$1}'
+ filesize=8
+ checkdiskspace bizcomp_users.json 8
+ location=bizcomp_users.json
+ KBrequired=8
+ '[' xbizcomp_users.json = x ']'
+ '[' x8 = x ']'
+ stripdir=bizcomp_users.json
++ df -P -k bizcomp_users.json
++ tail -n 1
++ awk '{print$4}'
++ sed -e 's/K$//'
+ KBavail=51638004
+ '[' 51638004 -ge 8 ']'
+ return 0
+ tr -d '\r'
+ '[' 0 = 0 ']'
+ mv bizcomp_users.json.tmp bizcomp_users.json
+ '[' 0 = 0 ']'
+ echo '... INFO: Completed successfully.'
... INFO: Completed successfully.
+ echo '... INFO: Amending file to make it suitable for Import.'
... INFO: Amending file to make it suitable for Import.
+ echo '... INFO: Stage 1 - Document filtering'
... INFO: Stage 1 - Document filtering
++ du -P -k bizcomp_users.json
++ awk '{print$1}'
+ '[' 8 -ge 256000 ']'
++ wc -l bizcomp_users.json
++ awk '{print$1}'
+ KBreduction=1
++ du -P -k bizcomp_users.json
++ awk '{print$1}'
+ filesize=8
++ expr 8 - 1
+ filesize=7
+ checkdiskspace bizcomp_users.json 7
+ location=bizcomp_users.json
+ KBrequired=7
+ '[' xbizcomp_users.json = x ']'
+ '[' x7 = x ']'
+ stripdir=bizcomp_users.json
++ df -P -k bizcomp_users.json
++ tail -n 1
++ awk '{print$4}'
++ sed -e 's/K$//'
+ KBavail=51638004
+ '[' 51638004 -ge 7 ']'
+ return 0
+ sed -i '""' 's/.*,"doc"://g' bizcomp_users.json
+ '[' '!' 0 = 0 ']'
+ echo '... INFO: Stage 2 - Duplicate curly brace removal'
... INFO: Stage 2 - Duplicate curly brace removal
++ wc -l bizcomp_users.json
++ awk '{print$1}'
+ KBreduction=0
++ du -P -k bizcomp_users.json
++ awk '{print$1}'
+ filesize=8
++ expr 8 - 0
+ filesize=8
+ checkdiskspace bizcomp_users.json 8
+ location=bizcomp_users.json
+ KBrequired=8
+ '[' xbizcomp_users.json = x ']'
+ '[' x8 = x ']'
+ stripdir=bizcomp_users.json
++ df -P -k bizcomp_users.json
++ tail -n 1
++ awk '{print$4}'
++ sed -e 's/K$//'
+ KBavail=51637996
+ '[' 51637996 -ge 8 ']'
+ return 0
+ sed -i '""' 's/}},$/},/g' bizcomp_users.json
+ '[' '!' 0 = 0 ']'
+ echo '... INFO: Stage 3 - Header Correction'
... INFO: Stage 3 - Header Correction
++ du -P -k bizcomp_users.json
++ awk '{print$1}'
+ filesize=8
+ checkdiskspace bizcomp_users.json 8
+ location=bizcomp_users.json
+ KBrequired=8
+ '[' xbizcomp_users.json = x ']'
+ '[' x8 = x ']'
+ stripdir=bizcomp_users.json
++ df -P -k bizcomp_users.json
++ tail -n 1
++ awk '{print$4}'
++ sed -e 's/K$//'
+ KBavail=51637996
+ '[' 51637996 -ge 8 ']'
+ return 0
+ sed -i '""' '1s/^.*/{"new_edits":false,"docs":[/' bizcomp_users.json
+ '[' '!' 0 = 0 ']'
+ echo '... INFO: Stage 4 - Final document line correction'
... INFO: Stage 4 - Final document line correction
++ du -P -k bizcomp_users.json
++ awk '{print$1}'
+ filesize=8
+ checkdiskspace bizcomp_users.json 8
+ location=bizcomp_users.json
+ KBrequired=8
+ '[' xbizcomp_users.json = x ']'
+ '[' x8 = x ']'
+ stripdir=bizcomp_users.json
++ df -P -k bizcomp_users.json
++ tail -n 1
++ awk '{print$4}'
++ sed -e 's/K$//'
+ KBavail=51637996
+ '[' 51637996 -ge 8 ']'
+ return 0
+ sed -i '""' 's/}}$/}/g' bizcomp_users.json
+ '[' '!' 0 = 0 ']'
+ echo '... INFO: Export completed successfully. File available at: bizcomp_users.json'
... INFO: Export completed successfully. File available at: bizcomp_users.json

**Comment by pjatinsight** [comment]
Was that any use to you?

**Comment by dalgibbard** [comment]
It was indeed. I'm guessing at the moment, but it might be caused by the recent changes to the sed statements. I'll have to take a look when I have a Mac to hand :)

**Comment by pjatinsight** [comment]
Ok,
Well, if there is anything more I can do to help then let me know.
pj

——————————————

PJ Mealy
pj.mealy@insight-centre.org <mailto:pj.mealy@insight-centre.org>

Insight Centre for Data Analytics
Galway Business Park
Dangan
Galway
H91 AEX4

**Comment by dalgibbard** [comment]
Have recreated the issue here:
```
$ diff -u bizcomp_users.json*
--- bizcomp_users.json  2015-12-02 15:04:42.000000000 +0000
+++ bizcomp_users.json""        2015-12-02 15:04:42.000000000 +0000
@@ -4,5 +4,5 @@
 {"_id":"e164bb3745089d306cc10c6d1f001c67","_rev":"1-3aba33e0926a98fb0a7f87d7821b85b9","test2":"hello2"},
 {"_id":"e164bb3745089d306cc10c6d1f002668","_rev":"1-b8e58364840c56a2cb11641fe381ca06","test3":"hello3"},
 {"_id":"e164bb3745089d306cc10c6d1f00342e","_rev":"1-509070ee39c56445690185d3d6fa7b0e","test4":"hello4"},
-{"_id":"e164bb3745089d306cc10c6d1f003e02","_rev":"1-28410f25bcab4d3aa732d12dcc2e45ce","test5":"hello5"}
+{"_id":"e164bb3745089d306cc10c6d1f003e02","_rev":"1-28410f25bcab4d3aa732d12dcc2e45ce","test5":"hello5"}}
 ]}
```

So looks like the file with the extra quotes is prior to completion of sed processing, and is as a result of the new sed statements. The file without the quotes is the final json (and is actually valid, whereas the quotes file is not valid JSON).
Looking into it.

**Comment by dalgibbard** [comment]
annoyingly, bash is substituting it in an odd manner - when swapped out for single quotes:
```sed '-i'\'''\''' 's/}}$/}/g' bizcomp_users.json```
looking for a suitable workaround.

**Comment by dalgibbard** [comment]
Have committed a fix for this, and upped version to 1.1.3.
Opted to specify a real suffix to the files, and simply remove the file after the sed is complete (essentially the same behaviour as 'sed -i' in linux, but without the filename suffix being random.)
It's not as pretty as I'd like it to be, but hey- it works :)

Tested on Ubuntu 15.04 and Mac OSX El Capitan.

**Comment by pjatinsight** [comment]
Brilliant,
Thanks very much!
PJ
